sequencer/src/persistence/fs.rs

```diff
@@ -53,6 +64,113 @@ impl Persistence {
     fn anchor_leaf_path(&self) -> PathBuf {
         self.0.join("anchor_leaf")
     }
+
+    fn vid_dir_path(&self) -> PathBuf {
+        self.0.join("vid")
+    }
+
+    fn da_dir_path(&self) -> PathBuf {
+        self.0.join("da")
+    }
+
+    fn high_qc_path(&self) -> PathBuf {
+        self.0.join("high_qc")
+    }
+}
+
+#[async_trait]
+impl Storage<SeqTypes> for Persistence {
+    async fn append_vid(
+        &self,
+        proposal: &Proposal<SeqTypes, VidDisperseShare<SeqTypes>>,
+    ) -> anyhow::Result<()> {
+        let view_number = proposal.data.get_view_number().get_u64();
+        let dir_path = self.vid_dir_path();
+
+        fs::create_dir_all(dir_path.clone()).context("failed to create vid dir")?;
+
+        let file_path = dir_path.join(view_number.to_string()).with_extension("txt");
+
+        let mut file = fs::OpenOptions::new()
+            .append(true)
+            .create(true)
+            .open(file_path)?;
+
+        let proposal_bytes = bincode::serialize(&proposal).context("serialize proposal")?;
+
+        file.write_all(&proposal_bytes)?;
+
+        Ok(())
+    }
+    async fn append_da(
+        &self,
+        proposal: &Proposal<SeqTypes, DAProposal<SeqTypes>>,
+    ) -> anyhow::Result<()> {
+        let view_number = proposal.data.get_view_number().get_u64();
+        let dir_path = self.da_dir_path();
+
+        fs::create_dir_all(dir_path.clone()).context("failed to create da dir")?;
+
+        let file_path = dir_path.join(view_number.to_string()).with_extension("txt");
+
+        let mut file = fs::OpenOptions::new()
+            .append(true)
+            .create(true)
+            .open(file_path)?;
+
+        let proposal_bytes = bincode::serialize(&proposal).context("serialize proposal")?;
+
+        file.write_all(&proposal_bytes)?;
+
+        Ok(())
+    }
+    async fn record_action(&self, view: ViewNumber, action: HotShotAction) -> anyhow::Result<()> {
+        if let HotShotAction::Vote = action {
```

Comment:
Ah, so the `voted_view` stuff was from before we realized that there were other attributable actions. We want to store the latest view in which we have taken _any_ attributable action, so we should just store the view number here regardless of the type of action, and change `load_voted_view` to `load_latest_acted_view` or something